### ISSUE TITLE
[Bug] Repeated "save login" prompt after logging in on TreeHerder

### ISSUE THREAD

**Issue by staktrace** [mozilla-mobile/fenix#8786]
## Steps to reproduce

1. Open a tab with https://news.ycombinator.com/ (this step may not be required, but I had it open while finding steps for a different bug so I'm including it here for completeness)
2. Open a new tab and navigate to https://treeherder.mozilla.org
3. Click on the log in button on the top right
4. Follow the login flow as prompted (you have to enter Mozilla LDAP/2FA credentials, accept the taskcluster prompt, etc.) until you are back to treeherder and logged in.
5. At this point Fenix should prompt you "Do you want Firefox Nightly to save this login?" Click "Don't save"
6. Hit the tab counter to go to the tabs list
7. Re-select the treeherder tab

### Expected behavior

You see the treeherder tab and everything is a-ok.

### Actual behavior

Fenix again prompts "Do you want Firefox Nightly to save this login?". In fact even if you go into Fenix settings and then go back to the tab, it will prompt you again. switch tabs and switch back? Prompt again!

### Device information

* Android device: Sony Xperia XZ1C, Android 9
* Fenix version: Nightly 200226 06:01 (Build #20570605) 34.0.0, 53f1356d7 GV: 75.0a1-20200225094028

**Comment by pdebruine** [comment]
I came across the same issue with https://haveibeenpwned.com/Passwords. Steps to reproduce:

1. Navigate to https://haveibeenpwned.com/Passwords
2. Enter a password and search
3. In the save password prompt choose Don't save
4. Navigate to a different site (google.com or something)
5. The prompt to save the password for haveibeenpwnd.com is shown again
6. Switch to a different tab then go back to the original tab, the prompt is shown again

**Expected behaviour**
No prompt for an old login when I already indicated to not save

**Actual behaviour**
Firefox/Fenix shows the prompt multiple times even when no longer visiting the site the login was originally entered for.
